packages/yoastseo/src/parse/build/private/parseBlocks.js

```diff
@@ -39,21 +40,33 @@ function updateBlocksOffset( blocks, text ) {
 	if ( blocks.length === 0 ) {
 		return;
 	}
-	blocks.forEach( currentBlock => {
+	blocks.forEach( ( currentBlock, index ) => {
 		const matches = blockTokenizer.exec( text );
-
-		if ( null === matches ) {
-			return;
+		if ( currentBlock.name === "core/freeform" ) {
+			const previousBlock = blocks[ index - 1 ];
+			if ( previousBlock ) {
+				const previousBlockEndOffset = previousBlock.endOffset;
+				const startedAt = previousBlockEndOffset + 2;
+				currentBlock.startOffset = startedAt;
+				currentBlock.endOffset = startedAt + currentBlock.blockLength;
+				currentBlock.contentOffset = startedAt;
+			} else {
+				currentBlock.startOffset = 0;
+				currentBlock.endOffset = 0 + currentBlock.blockLength;
+				currentBlock.contentOffset = 0;
+			}
+		} else {
+			if ( null === matches ) {
+				return;
+			}
+			const [ match ] = matches;
+			const startedAt = matches.index;
+			const length = match.length;
+			currentBlock.startOffset = startedAt;
+			currentBlock.endOffset = startedAt + currentBlock.blockLength;
+			currentBlock.contentOffset = startedAt + length + 1;
```

Comment:
I was thinking, wouldn't it actually be better to do the blocks mapping (adding the `cotentOffset`, `startOffset`, and `endOffset` ) when we collect the analysis data (in `collectAnalysisData.js`)? 🤔 
This way we make `yoastseo` a little bit less platform-dependent.
@mhkuu or @mykola what do you think?